Description: Too long hostnames and schemes are no longer fatal errors
 proxy_util: Schemes and hostnames that are "too long" are
 no longer automatically fatal errors but are instead logged
 and truncated, at which point the admin can determine if that
 is OK or not.
 trunk patch: http://svn.apache.org/r1823482
 +1: jim, minfrin, rpluem
Origin: upstream, http://svn.apache.org/viewvc?view=revision&revision=1824455
Bug: https://bz.apache.org/bugzilla/show_bug.cgi?id=62085
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/apache2/+bug/1750356
Last-Update: 2018-06-27
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- 2.4.x/modules/proxy/proxy_util.c	2018/02/16 12:31:28	1824454
+++ 2.4.x/modules/proxy/proxy_util.c	2018/02/16 12:35:03	1824455
@@ -1691,10 +1691,12 @@
         "Alert! worker name (%s) too long; truncated to: %s", ptr, wshared->name);
     }
     if (PROXY_STRNCPY(wshared->scheme, uri.scheme) != APR_SUCCESS) {
-        return apr_psprintf(p, "worker scheme (%s) too long", uri.scheme);
+        ap_log_error(APLOG_MARK, APLOG_ERR, 0, ap_server_conf, APLOGNO(010117)
+        "Alert! worker scheme (%s) too long; truncated to: %s", uri.scheme, wshared->scheme);
     }
     if (PROXY_STRNCPY(wshared->hostname, uri.hostname) != APR_SUCCESS) {
-        return apr_psprintf(p, "worker hostname (%s) too long", uri.hostname);
+        ap_log_error(APLOG_MARK, APLOG_ERR, 0, ap_server_conf, APLOGNO(010118)
+        "Alert! worker hostname (%s) too long; truncated to: %s", uri.hostname, wshared->hostname);
     }
     wshared->flush_packets = flush_off;
     wshared->flush_wait = PROXY_FLUSH_WAIT;
